TypewiseAlert/TypewiseAlert.cs

```diff
@@ -1,90 +1,180 @@
-        case AlertTarget.TO_EMAIL:
-          sendToEmail(breachType);
-          break;
-      }
-    }
-    public static void sendToController(BreachType breachType) {
-      const ushort header = 0xfeed;
-      Console.WriteLine("{} : {}\n", header, breachType);
-    }
-    public static void sendToEmail(BreachType breachType) {
-      string recepient = "a.b@c.com";
-      switch(breachType) {
-        case BreachType.TOO_LOW:
-          Console.WriteLine("To: {}\n", recepient);
-          Console.WriteLine("Hi, the temperature is too low\n");
-          break;
-        case BreachType.TOO_HIGH:
-          Console.WriteLine("To: {}\n", recepient);
-          Console.WriteLine("Hi, the temperature is too high\n");
-          break;
-        case BreachType.NORMAL:
-          break;
-      }
+    public class TypewiseAlert
+    {       
+        public enum BreachType {
+            NORMAL,
+            TOO_LOW,
+            TOO_HIGH
+        };
+        
+        public enum CoolingType
+        {
+            PASSIVE_COOLING,
+            HI_ACTIVE_COOLING,
+            MED_ACTIVE_COOLING
+        };
+
+        public enum AlertTarget
+        {
+            TO_CONTROLLER,
+            TO_EMAIL
+        };
+
+        public class CoolingLimitDictionaryInitializer
+        {
+            public Dictionary<CoolingType, Func<ILimitInitializer>> _CoolingLimitType = new Dictionary<CoolingType, Func<ILimitInitializer>>();
+
+            public CoolingLimitDictionaryInitializer()
+            {
+                _CoolingLimitType.Add(CoolingType.PASSIVE_COOLING, () => { return new PassiveCoolingLimit(); });
+                _CoolingLimitType.Add(CoolingType.HI_ACTIVE_COOLING, () => { return new HighCoolingLimit(); });
+                _CoolingLimitType.Add(CoolingType.MED_ACTIVE_COOLING, () => { return new MediumCoolingLimit(); });
+            }
+        }
+
+        public class EmailMessageInitializer
+        {
+            public Dictionary<BreachType, Func<IEmailTrigger>> _Email = new Dictionary<BreachType, Func<IEmailTrigger>>();
+
+            public EmailMessageInitializer()
+            {
+                _Email.Add(BreachType.TOO_HIGH, () => { return new HighLimitMessageEmail(); });
+                _Email.Add(BreachType.TOO_LOW, () => { return new LowLimitMessageEmail(); });
+                _Email.Add(BreachType.NORMAL, () => { return new NormalLimitMessageEmail(); });
+            }
+        }
+
+        public class AlertTargetType
+        {
+            public Dictionary<AlertTarget, String> _AlertTargetType = new Dictionary<AlertTarget, String>();
+            public AlertTargetType(BreachType breachType)
+            {
+                _AlertTargetType.Add(AlertTarget.TO_CONTROLLER, SendToController(breachType));
+                _AlertTargetType.Add(AlertTarget.TO_EMAIL, SendToEmail(breachType));
+            }
+        }
+
+        public static BreachType InferBreach(double value, double lowerLimit, double upperLimit) {
```

Comment:
InferBreach still open for Modification

Reply:
Replaced If-else conditions with ternary operator

Reply:
How does ternary operator will resolve OCP issues?